(WIP) add information about the end date of plan into Order

PetrDlouhy opened the pull request:
This PR is not ready yet. In current state it is meant to be basis for further discussion.

In my project, I would need to redistribute money from users paying for the subscription plans to the creators of content. The current state of `django-plans` model is quite insufficient for this. It doesn't capture complete log of plans usage (from when to when was the `Order` used extend `Plan`.

This PR is my attempt to extend the model to be able to capture this. Please, comment if you think that this is in the right direction, or what would you recommend. I am not sure if it would work properly if there is a change of plan in middle the extension time.

Alir3z4 commented:
First and foremost, django-plans doesn't support subscription plans yet https://github.com/django-getpaid/django-plans/issues/64

Does your work will include that as well ?

PetrDlouhy commented:
@Alir3z4 No, I meant the plan order by subscription. This only enables tracking of history of plan renewals in current logic.

PetrDlouhy commented:
I think this is more or less complete, although it would need some review. I am removing the Work in progress status.

PetrDlouhy commented:
@Alir3z4 I added Changelog record and fixed the `help_text` problem requested in #82 

Note: the migration would clash with migrations in #80 and #79, so I will need to rename the other migrations when one of these PRs get pulled.
